src/distilabel/pipeline.py

```diff
@@ -611,26 +612,51 @@ def _generate(  # noqa: C901
                         labels.append(batch_labels)  # type: ignore
                     else:
                         labels.extend(batch_labels)  # type: ignore
+
                 except Exception as e:
-                    if not enable_checkpoints:
+                    if not checkpoint_strategy:
                         raise RuntimeError(
-                            "`Pipeline.generate` failed during labelling step. Setting `enable_checkpoints=True` is recommended!"
+                            "`Pipeline.generate` failed during labelling step. Passing a `DatasetCheckpoint` is recommended!"
                         ) from e
                     logger.error(
                         f"`Pipeline.generate` failed during labelling step with exception: {e}"
                     )
+
                     return self._build_dataset(
                         dataset,
                         generations=generations,
                         labels=labels,
                         batch_size=batch_size,
                     )
 
+            if checkpoint_strategy and checkpoint_strategy.do_checkpoint(
+                batch_i * batch_size
+            ):
+                logger.info(f"Saving dataset up to batch {batch_i}...")
```

Comment:
Would it useful for the user to print the writing file here?

Reply:
do you mean the path to the file? Will do it!